avocado/core/runner.py

```diff
@@ -468,6 +469,12 @@ def _iter_variants(template, mux):
         :raises ValueError: When variant and template declare params.
         """
         for variant, params in mux.itertests():
+            variant_name = ''
+            for x in params[0]:
+                path = x.path[len('/run'):]
+                variant_name += path.replace('/', '-')
+            # trim first '-'
+            variant_name = variant_name[1:]
```

Comment:
One more think, I liked the concept of short-id. How about adding `!shortname` tag (or just `!variant` or `!id`) to yaml and allow constructing the variant id from children of such nodes? And if we define this on the `Mux` level, we'd avoid the need to implement it per-plugin and it'd simple be there.